sdk/ml/azure-ai-ml/azure/ai/ml/exceptions.py

```diff
@@ -214,17 +212,15 @@ def __init__(
         message: str,
         no_personal_data_message: str,
         *args,
-        target: ErrorTarget = ErrorTarget.UNKNOWN,
-        error_category: ErrorCategory = ErrorCategory.UNKNOWN,
+        target: Union[str, ErrorTarget] = ErrorTarget.UNKNOWN,
+        error_category: Union[str, ErrorTarget] = ErrorCategory.UNKNOWN,
         **kwargs,
     ):
         super(ComponentException, self).__init__(
             message=message,
+            no_personal_data_message=no_personal_data_message,
             target=target,
             error_category=error_category,
-            no_personal_data_message=no_personal_data_message,
```

Comment:
why remove the args and kwargs?

Reply:
![image](https://github.com/Azure/azure-sdk-for-python/assets/19650434/0415164b-0353-4935-b2f8-8212449295b5)

mypy would complain about this error

Reply:
Let's ignore those mypy errors for now.

Reply:
fixed